[CI] HistoryActionConditionTests.testActionCondition is failing

**Build scan**: https://gradle-enterprise.elastic.co/s/qhcmhs7bgvyyq

**Repro line**:
```
./gradlew ':x-pack:plugin:watcher:internalClusterTest' --tests "org.elasticsearch.xpack.watcher.history.HistoryActionConditionTests.testActionCondition" -Dtests.seed=3837630D03B63CCA -Dtests.security.manager=true -Dtests.locale=es-DO -Dtests.timezone=America/Argentina/Catamarca -Druntime.java=11
```

**Reproduces locally?**: No

**Applicable branches**: `master`

**Failure history**:
 [build stats](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-30d,mode:quick,to:now))&_a=(columns:!(_source),index:b646ed00-7efc-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:'%22HistoryActionConditionTests%20testActionCondition%22'),sort:!(process.time-start,desc))) 
 Failed 16 times in the past month

**Failure excerpt**:
```
org.elasticsearch.xpack.watcher.history.HistoryActionConditionTests > testActionCondition FAILED
    java.lang.AssertionError: watch was triggered on [0] schedulers, expected [1]
    Expected: a value equal to or greater than <1L>
         but: <0L> was less than <1L>
        at __randomizedtesting.SeedInfo.seed([3837630D03B63CCA:C89F01B865DB514C]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.elasticsearch.xpack.watcher.test.AbstractWatcherIntegrationTestCase$TimeWarp.trigger(AbstractWatcherIntegrationTestCase.java:606)
        at org.elasticsearch.xpack.watcher.test.AbstractWatcherIntegrationTestCase$TimeWarp.trigger(AbstractWatcherIntegrationTestCase.java:591)
        at org.elasticsearch.xpack.watcher.history.HistoryActionConditionTests.putAndTriggerWatch(HistoryActionConditionTests.java:259)
        at org.elasticsearch.xpack.watcher.history.HistoryActionConditionTests.testActionCondition(HistoryActionConditionTests.java:196)
REPRODUCE WITH: ./gradlew ':x-pack:plugin:watcher:internalClusterTest' --tests "org.elasticsearch.xpack.watcher.history.HistoryActionConditionTests.testActionCondition" -Dtests.seed=3837630D03B63CCA -Dtests.security.manager=true -Dtests.locale=es-DO -Dtests.timezone=America/Argentina/Catamarca -Druntime.java=11

Suite: Test class org.elasticsearch.xpack.watcher.history.HistoryActionConditionTests
  1> [2020-11-16T07:57:11,025][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionConditionWithFailures] before test
  1> [2020-11-16T07:57:11,028][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionConditionWithFailures] [HistoryActionConditionTests#testActionConditionWithFailures]: setting up test
  1> [2020-11-16T07:57:11,030][INFO ][o.e.t.InternalTestCluster] [testActionConditionWithFailures] Setup InternalTestCluster [SUITE-TEST_WORKER_VM=[587]-CLUSTER_SEED=[-4459053671090325947]-HASH=[13E62C17086]-cluster] with seed [C21E424FAC427645] using [0] dedicated masters, [2] (data) nodes and [0] coord only nodes (master nodes are [auto-managed])
  1> [2020-11-16T07:57:11,052][INFO ][o.e.n.Node               ] [testActionConditionWithFailures] version[8.0.0-SNAPSHOT], pid[139857], build[unknown/unknown/62e1d113b7153180c4a68af330629d388bef4b7a/2020-11-16T10:35:31.960618Z], OS[Linux/4.18.0-193.28.1.el8_2.x86_64/amd64], JVM[Oracle Corporation/OpenJDK 64-Bit Server VM/11.0.2/11.0.2+7]
  1> [2020-11-16T07:57:11,052][INFO ][o.e.n.Node               ] [testActionConditionWithFailures] JVM home [/var/lib/jenkins/.java/openjdk-11.0.2-linux]
  1> [2020-11-16T07:57:11,052][DEPRECATION][o.e.d.n.Node             ] [testActionConditionWithFailures] data_stream.dataset="deprecation.elasticsearch" data_stream.namespace="default" data_stream.type="logs" ecs.version="1.6" key="no-jdk" message="no-jdk distributions that do not bundle a JDK are deprecated and will be removed in a future release"
  1> [2020-11-16T07:57:11,053][INFO ][o.e.n.Node               ] [testActionConditionWithFailures] JVM arguments [-Dfile.encoding=UTF8, -Des.scripting.update.ctx_in_params=false, -Des.search.rewrite_sort=true, -Des.set.netty.runtime.available.processors=false, -Des.transport.cname_in_publish_address=true, -Dgradle.dist.lib=/var/lib/jenkins/.gradle/wrapper/dists/gradle-6.6.1-all/ejrtlte9hlw8v6ii20a9584rs/gradle-6.6.1/lib, -Dgradle.user.home=/var/lib/jenkins/.gradle, -Dgradle.worker.jar=/var/lib/jenkins/.gradle/caches/6.6.1/workerMain/gradle-worker.jar, -Dio.netty.noKeySetOptimization=true, -Dio.netty.noUnsafe=true, -Dio.netty.recycler.maxCapacityPerThread=0, -Djava.awt.headless=true, -Djava.locale.providers=SPI,COMPAT, -Djna.nosys=true, -Dorg.gradle.native=false, -Dtests.artifact=watcher, -Dtests.gradle=true, -Dtests.logger.level=WARN, -Dtests.security.manager=true, -Dtests.seed=3837630D03B63CCA, -Dtests.task=:x-pack:plugin:watcher:internalClusterTest, --illegal-access=warn, -XX:+HeapDumpOnOutOfMemoryError, -esa, -XX:HeapDumpPath=/dev/shm/elastic+elasticsearch+master+multijob+fast+part2/x-pack/plugin/watcher/build/heapdump, -Xms512m, -Xmx512m, -Dfile.encoding=UTF-8, -Djava.io.tmpdir=/dev/shm/elastic+elasticsearch+master+multijob+fast+part2/x-pack/plugin/watcher/build/testrun/internalClusterTest/temp, -Duser.country=US, -Duser.language=en, -Duser.variant, -ea]
  1> [2020-11-16T07:57:11,053][WARN ][o.e.n.Node               ] [testActionConditionWithFailures] version [8.0.0-SNAPSHOT] is a pre-release version of Elasticsearch and is not suitable for production
  1> [2020-11-16T07:57:11,056][INFO ][o.e.x.w.t.TimeWarpedWatcher] [testActionConditionWithFailures] using time warped watchers plugin
  1> [2020-11-16T07:57:11,059][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] no modules loaded
  1> [2020-11-16T07:57:11,060][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.analysis.common.CommonAnalysisPlugin]
  1> [2020-11-16T07:57:11,060][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.index.MockEngineFactoryPlugin]
  1> [2020-11-16T07:57:11,060][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.index.mapper.MockFieldFilterPlugin]
  1> [2020-11-16T07:57:11,060][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.script.MockMustacheScriptEngine$TestPlugin]
  1> [2020-11-16T07:57:11,060][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.script.MockScriptService$TestPlugin]
  1> [2020-11-16T07:57:11,060][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.test.ESIntegTestCase$AssertActionNamePlugin]
  1> [2020-11-16T07:57:11,060][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.test.ESIntegTestCase$TestSeedPlugin]
  1> [2020-11-16T07:57:11,060][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.test.MockHttpTransport$TestPlugin]
  1> [2020-11-16T07:57:11,060][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.test.TestGeoShapeFieldMapperPlugin]
  1> [2020-11-16T07:57:11,060][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.test.store.MockFSIndexStore$TestPlugin]
  1> [2020-11-16T07:57:11,061][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.transport.nio.MockNioTransportPlugin]
  1> [2020-11-16T07:57:11,061][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.xpack.datastreams.DataStreamsPlugin]
  1> [2020-11-16T07:57:11,061][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.xpack.ilm.IndexLifecycle]
  1> [2020-11-16T07:57:11,061][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.xpack.watcher.history.HistoryActionConditionTests$CustomScriptPlugin]
  1> [2020-11-16T07:57:11,061][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.xpack.watcher.test.TimeWarpedWatcher]
  1> [2020-11-16T07:57:11,065][INFO ][o.e.e.NodeEnvironment    ] [testActionConditionWithFailures] using [1] data paths, mounts [[/dev/shm (tmpfs)]], net usable_space [43.7gb], net total_space [47gb], types [tmpfs]
  1> [2020-11-16T07:57:11,065][INFO ][o.e.e.NodeEnvironment    ] [testActionConditionWithFailures] heap size [512mb], compressed ordinary object pointers [true]
  1> [2020-11-16T07:57:11,067][INFO ][o.e.n.Node               ] [testActionConditionWithFailures] node name [node_s0], node ID [3zTZs07tSG2eYy_GkUgUUw], cluster name [SUITE-TEST_WORKER_VM=[587]-CLUSTER_SEED=[-4459053671090325947]-HASH=[13E62C17086]-cluster], roles [data_hot, data_content, ingest, data_warm, data, data_cold, master, remote_cluster_client]
  1> [2020-11-16T07:57:11,283][INFO ][o.e.d.DiscoveryModule    ] [testActionConditionWithFailures] using discovery type [zen] and seed hosts providers [settings, file]
  1> [2020-11-16T07:57:11,367][INFO ][o.e.n.Node               ] [testActionConditionWithFailures] initialized
  1> [2020-11-16T07:57:11,370][INFO ][o.e.n.Node               ] [testActionConditionWithFailures] version[8.0.0-SNAPSHOT], pid[139857], build[unknown/unknown/62e1d113b7153180c4a68af330629d388bef4b7a/2020-11-16T10:35:31.960618Z], OS[Linux/4.18.0-193.28.1.el8_2.x86_64/amd64], JVM[Oracle Corporation/OpenJDK 64-Bit Server VM/11.0.2/11.0.2+7]
  1> [2020-11-16T07:57:11,370][INFO ][o.e.n.Node               ] [testActionConditionWithFailures] JVM home [/var/lib/jenkins/.java/openjdk-11.0.2-linux]
  1> [2020-11-16T07:57:11,370][DEPRECATION][o.e.d.n.Node             ] [testActionConditionWithFailures] data_stream.dataset="deprecation.elasticsearch" data_stream.namespace="default" data_stream.type="logs" ecs.version="1.6" key="no-jdk" message="no-jdk distributions that do not bundle a JDK are deprecated and will be removed in a future release"
  1> [2020-11-16T07:57:11,371][INFO ][o.e.n.Node               ] [testActionConditionWithFailures] JVM arguments [-Dfile.encoding=UTF8, -Des.scripting.update.ctx_in_params=false, -Des.search.rewrite_sort=true, -Des.set.netty.runtime.available.processors=false, -Des.transport.cname_in_publish_address=true, -Dgradle.dist.lib=/var/lib/jenkins/.gradle/wrapper/dists/gradle-6.6.1-all/ejrtlte9hlw8v6ii20a9584rs/gradle-6.6.1/lib, -Dgradle.user.home=/var/lib/jenkins/.gradle, -Dgradle.worker.jar=/var/lib/jenkins/.gradle/caches/6.6.1/workerMain/gradle-worker.jar, -Dio.netty.noKeySetOptimization=true, -Dio.netty.noUnsafe=true, -Dio.netty.recycler.maxCapacityPerThread=0, -Djava.awt.headless=true, -Djava.locale.providers=SPI,COMPAT, -Djna.nosys=true, -Dorg.gradle.native=false, -Dtests.artifact=watcher, -Dtests.gradle=true, -Dtests.logger.level=WARN, -Dtests.security.manager=true, -Dtests.seed=3837630D03B63CCA, -Dtests.task=:x-pack:plugin:watcher:internalClusterTest, --illegal-access=warn, -XX:+HeapDumpOnOutOfMemoryError, -esa, -XX:HeapDumpPath=/dev/shm/elastic+elasticsearch+master+multijob+fast+part2/x-pack/plugin/watcher/build/heapdump, -Xms512m, -Xmx512m, -Dfile.encoding=UTF-8, -Djava.io.tmpdir=/dev/shm/elastic+elasticsearch+master+multijob+fast+part2/x-pack/plugin/watcher/build/testrun/internalClusterTest/temp, -Duser.country=US, -Duser.language=en, -Duser.variant, -ea]
  1> [2020-11-16T07:57:11,371][WARN ][o.e.n.Node               ] [testActionConditionWithFailures] version [8.0.0-SNAPSHOT] is a pre-release version of Elasticsearch and is not suitable for production
  1> [2020-11-16T07:57:11,371][INFO ][o.e.x.w.t.TimeWarpedWatcher] [testActionConditionWithFailures] using time warped watchers plugin
  1> [2020-11-16T07:57:11,372][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] no modules loaded
  1> [2020-11-16T07:57:11,372][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.analysis.common.CommonAnalysisPlugin]
  1> [2020-11-16T07:57:11,372][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.index.MockEngineFactoryPlugin]
  1> [2020-11-16T07:57:11,372][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.index.mapper.MockFieldFilterPlugin]
  1> [2020-11-16T07:57:11,372][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.script.MockMustacheScriptEngine$TestPlugin]
  1> [2020-11-16T07:57:11,372][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.script.MockScriptService$TestPlugin]
  1> [2020-11-16T07:57:11,372][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.test.ESIntegTestCase$AssertActionNamePlugin]
  1> [2020-11-16T07:57:11,372][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.test.ESIntegTestCase$TestSeedPlugin]
  1> [2020-11-16T07:57:11,373][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.test.MockHttpTransport$TestPlugin]
  1> [2020-11-16T07:57:11,384][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.test.TestGeoShapeFieldMapperPlugin]
  1> [2020-11-16T07:57:11,384][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.test.store.MockFSIndexStore$TestPlugin]
  1> [2020-11-16T07:57:11,384][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.transport.nio.MockNioTransportPlugin]
  1> [2020-11-16T07:57:11,384][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.xpack.datastreams.DataStreamsPlugin]
  1> [2020-11-16T07:57:11,384][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.xpack.ilm.IndexLifecycle]
  1> [2020-11-16T07:57:11,384][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.xpack.watcher.history.HistoryActionConditionTests$CustomScriptPlugin]
  1> [2020-11-16T07:57:11,385][INFO ][o.e.p.PluginsService     ] [testActionConditionWithFailures] loaded plugin [org.elasticsearch.xpack.watcher.test.TimeWarpedWatcher]
  1> [2020-11-16T07:57:11,390][INFO ][o.e.e.NodeEnvironment    ] [testActionConditionWithFailures] using [1] data paths, mounts [[/dev/shm (tmpfs)]], net usable_space [43.7gb], net total_space [47gb], types [tmpfs]
  1> [2020-11-16T07:57:11,390][INFO ][o.e.e.NodeEnvironment    ] [testActionConditionWithFailures] heap size [512mb], compressed ordinary object pointers [true]
  1> [2020-11-16T07:57:11,392][INFO ][o.e.n.Node               ] [testActionConditionWithFailures] node name [node_s1], node ID [_ZCPAbBOT6mkCwG0a2vK2A], cluster name [SUITE-TEST_WORKER_VM=[587]-CLUSTER_SEED=[-4459053671090325947]-HASH=[13E62C17086]-cluster], roles [data_hot, data_content, ingest, data_warm, data, data_cold, master, remote_cluster_client]
  1> [2020-11-16T07:57:11,523][INFO ][o.e.d.DiscoveryModule    ] [testActionConditionWithFailures] using discovery type [zen] and seed hosts providers [settings, file]
  1> [2020-11-16T07:57:11,588][INFO ][o.e.n.Node               ] [testActionConditionWithFailures] initialized
  1> [2020-11-16T07:57:11,590][INFO ][o.e.n.Node               ] [[test_SUITE-TEST_WORKER_VM=[587]-CLUSTER_SEED=[-4459053671090325947]-HASH=[13E62C17086]-cluster[T#1]]] starting ...
  1> [2020-11-16T07:57:11,591][INFO ][o.e.n.Node               ] [[test_SUITE-TEST_WORKER_VM=[587]-CLUSTER_SEED=[-4459053671090325947]-HASH=[13E62C17086]-cluster[T#2]]] starting ...
  1> [2020-11-16T07:57:11,628][INFO ][o.e.t.TransportService   ] [[test_SUITE-TEST_WORKER_VM=[587]-CLUSTER_SEED=[-4459053671090325947]-HASH=[13E62C17086]-cluster[T#1]]] publish_address {127.0.0.1:46521}, bound_addresses {[::1]:36023}, {127.0.0.1:46521}
  1> [2020-11-16T07:57:11,633][INFO ][o.e.t.TransportService   ] [[test_SUITE-TEST_WORKER_VM=[587]-CLUSTER_SEED=[-4459053671090325947]-HASH=[13E62C17086]-cluster[T#2]]] publish_address {127.0.0.1:38539}, bound_addresses {[::1]:33097}, {127.0.0.1:38539}
  1> [2020-11-16T07:57:11,758][INFO ][o.e.c.c.Coordinator      ] [node_s1] setting initial configuration to VotingConfiguration{3zTZs07tSG2eYy_GkUgUUw,_ZCPAbBOT6mkCwG0a2vK2A}
  1> [2020-11-16T07:57:11,870][INFO ][o.e.c.s.MasterService    ] [node_s1] elected-as-master ([2] nodes joined)[{node_s1}{_ZCPAbBOT6mkCwG0a2vK2A}{pw7UKqdyQieFv3_ojvcfAA}{127.0.0.1}{127.0.0.1:38539}{cdhimrsw}{xpack.installed=true} elect leader, {node_s0}{3zTZs07tSG2eYy_GkUgUUw}{CtcC-6GpRuOl2bYCA2uWcg}{127.0.0.1}{127.0.0.1:46521}{cdhimrsw}{xpack.installed=true} elect leader, _BECOME_MASTER_TASK_, _FINISH_ELECTION_], term: 1, version: 1, delta: master node changed {previous [], current [{node_s1}{_ZCPAbBOT6mkCwG0a2vK2A}{pw7UKqdyQieFv3_ojvcfAA}{127.0.0.1}{127.0.0.1:38539}{cdhimrsw}{xpack.installed=true}]}, added {{node_s0}{3zTZs07tSG2eYy_GkUgUUw}{CtcC-6GpRuOl2bYCA2uWcg}{127.0.0.1}{127.0.0.1:46521}{cdhimrsw}{xpack.installed=true}}
  1> [2020-11-16T07:57:11,937][INFO ][o.e.c.c.CoordinationState] [node_s1] cluster UUID set to [VmVt4XeNSuy9Kh3nDOklmw]
  1> [2020-11-16T07:57:11,938][INFO ][o.e.c.c.CoordinationState] [node_s0] cluster UUID set to [VmVt4XeNSuy9Kh3nDOklmw]
  1> [2020-11-16T07:57:11,949][INFO ][o.e.c.s.ClusterApplierService] [node_s0] master node changed {previous [], current [{node_s1}{_ZCPAbBOT6mkCwG0a2vK2A}{pw7UKqdyQieFv3_ojvcfAA}{127.0.0.1}{127.0.0.1:38539}{cdhimrsw}{xpack.installed=true}]}, added {{node_s1}{_ZCPAbBOT6mkCwG0a2vK2A}{pw7UKqdyQieFv3_ojvcfAA}{127.0.0.1}{127.0.0.1:38539}{cdhimrsw}{xpack.installed=true}}, term: 1, version: 1, reason: ApplyCommitRequest{term=1, version=1, sourceNode={node_s1}{_ZCPAbBOT6mkCwG0a2vK2A}{pw7UKqdyQieFv3_ojvcfAA}{127.0.0.1}{127.0.0.1:38539}{cdhimrsw}{xpack.installed=true}}
  1> [2020-11-16T07:57:11,958][INFO ][o.e.n.Node               ] [[test_SUITE-TEST_WORKER_VM=[587]-CLUSTER_SEED=[-4459053671090325947]-HASH=[13E62C17086]-cluster[T#1]]] started
  1> [2020-11-16T07:57:11,971][INFO ][o.e.c.s.ClusterApplierService] [node_s1] master node changed {previous [], current [{node_s1}{_ZCPAbBOT6mkCwG0a2vK2A}{pw7UKqdyQieFv3_ojvcfAA}{127.0.0.1}{127.0.0.1:38539}{cdhimrsw}{xpack.installed=true}]}, added {{node_s0}{3zTZs07tSG2eYy_GkUgUUw}{CtcC-6GpRuOl2bYCA2uWcg}{127.0.0.1}{127.0.0.1:46521}{cdhimrsw}{xpack.installed=true}}, term: 1, version: 1, reason: Publication{term=1, version=1}
  1> [2020-11-16T07:57:12,019][INFO ][o.e.n.Node               ] [[test_SUITE-TEST_WORKER_VM=[587]-CLUSTER_SEED=[-4459053671090325947]-HASH=[13E62C17086]-cluster[T#2]]] started
  1> [2020-11-16T07:57:12,134][INFO ][o.e.g.GatewayService     ] [node_s1] recovered [0] indices into cluster_state
  1> [2020-11-16T07:57:12,167][INFO ][o.e.c.m.MetadataIndexTemplateService] [node_s1] adding index template [.watch-history-14] for index patterns [.watcher-history-14*]
  1> [2020-11-16T07:57:12,203][INFO ][o.e.c.m.MetadataIndexTemplateService] [node_s1] adding index template [.triggered_watches] for index patterns [.triggered_watches*]
  1> [2020-11-16T07:57:12,231][INFO ][o.e.c.m.MetadataIndexTemplateService] [node_s1] adding index template [.watches] for index patterns [.watches*]
  1> [2020-11-16T07:57:12,262][INFO ][o.e.c.m.MetadataIndexTemplateService] [node_s1] adding index template [.slm-history] for index patterns [.slm-history-5*]
  1> [2020-11-16T07:57:12,301][INFO ][o.e.x.i.a.TransportPutLifecycleAction] [node_s1] adding index lifecycle policy [watch-history-ilm-policy]
  1> [2020-11-16T07:57:12,514][INFO ][o.e.l.LicenseService     ] [node_s0] license [5e916730-c9c9-4733-a0f0-9e1f733b19a5] mode [trial] - valid
  1> [2020-11-16T07:57:12,553][INFO ][o.e.l.LicenseService     ] [node_s1] license [5e916730-c9c9-4733-a0f0-9e1f733b19a5] mode [trial] - valid
  1> [2020-11-16T07:57:12,555][INFO ][o.e.x.i.a.TransportPutLifecycleAction] [node_s1] adding index lifecycle policy [slm-history-ilm-policy]
  1> [2020-11-16T07:57:12,619][WARN ][o.e.c.m.MetadataIndexTemplateService] [node_s1] legacy template [random_index_template] has index patterns [*] matching patterns from existing composable templates [.watch-history-14,.triggered_watches,.slm-history,.watches] with patterns (.watch-history-14 => [.watcher-history-14*],.triggered_watches => [.triggered_watches*],.slm-history => [.slm-history-5*],.watches => [.watches*]); this template [random_index_template] may be ignored in favor of a composable template at index creation time
  1> [2020-11-16T07:57:12,621][INFO ][o.e.c.m.MetadataIndexTemplateService] [node_s1] adding template [random_index_template] for index patterns [*]
  1> [2020-11-16T07:57:12,658][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionConditionWithFailures] [HistoryActionConditionTests#testActionConditionWithFailures]: all set up test
  1> [2020-11-16T07:57:12,659][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionConditionWithFailures] [HistoryActionConditionTests#testActionConditionWithFailures]: freezing time on nodes
  1> [2020-11-16T07:57:12,662][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionConditionWithFailures] waiting to stop watcher, current states [Tuple [v1=node_s0 (0), v2=STARTED], Tuple [v1=node_s1 (0), v2=STARTED]]
  1> [2020-11-16T07:57:12,692][INFO ][o.e.x.w.WatcherService   ] [node_s0] stopping watch service, reason [watcher manually marked to shutdown by cluster state update]
  1> [2020-11-16T07:57:12,693][INFO ][o.e.x.w.WatcherLifeCycleService] [node_s0] watcher has stopped
  1> [2020-11-16T07:57:12,698][INFO ][o.e.x.w.WatcherService   ] [node_s1] stopping watch service, reason [watcher manually marked to shutdown by cluster state update]
  1> [2020-11-16T07:57:12,701][INFO ][o.e.x.w.WatcherLifeCycleService] [node_s1] watcher has stopped
  1> [2020-11-16T07:57:12,708][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionConditionWithFailures] waiting to stop watcher, current states [Tuple [v1=node_s0 (0), v2=STOPPED], Tuple [v1=node_s1 (0), v2=STOPPED]]
  1> [2020-11-16T07:57:12,716][DEPRECATION][o.e.d.c.m.MetadataCreateIndexService] [node_s1] data_stream.dataset="deprecation.elasticsearch" data_stream.namespace="default" data_stream.type="logs" ecs.version="1.6" key="index_name_starts_with_dot" message="index name [.watches] starts with a dot '.', in the next major version, index names starting with a dot are reserved for hidden indices and system indices"
  1> [2020-11-16T07:57:12,724][INFO ][o.e.c.m.MetadataCreateIndexService] [node_s1] [.watches] creating index, cause [api], templates [.watches], shards [3]/[0]
  1> [2020-11-16T07:57:12,725][INFO ][o.e.c.r.a.AllocationService] [node_s1] updating number_of_replicas to [1] for indices [.watches]
  1> [2020-11-16T07:57:13,086][DEPRECATION][o.e.d.c.m.MetadataCreateIndexService] [node_s1] data_stream.dataset="deprecation.elasticsearch" data_stream.namespace="default" data_stream.type="logs" ecs.version="1.6" key="index_name_starts_with_dot" message="index name [.triggered_watches-alias-index] starts with a dot '.', in the next major version, index names starting with a dot are reserved for hidden indices and system indices"
  1> [2020-11-16T07:57:13,121][INFO ][o.e.c.m.MetadataCreateIndexService] [node_s1] [.triggered_watches-alias-index] creating index, cause [Index to test aliases with .triggered-watches index], templates [.triggered_watches], shards [1]/[1]
  1> [2020-11-16T07:57:13,470][INFO ][o.e.c.m.MetadataCreateIndexService] [node_s1] [sasnccf] creating index, cause [api], templates [random_index_template], shards [1]/[1]
  1> [2020-11-16T07:57:13,910][INFO ][o.e.c.r.a.AllocationService] [node_s1] current.health="GREEN" message="Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[sasnccf][0]]])." previous.health="YELLOW" reason="shards started [[sasnccf][0]]"
  1> [2020-11-16T07:57:13,980][INFO ][o.e.c.m.MetadataDeleteIndexService] [node_s1] [.triggered_watches-alias-index/SUZPIm1fRoaBnpk_T2qfrw] deleting index
  1> [2020-11-16T07:57:14,092][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionConditionWithFailures] set alias for .triggered-watches index to [sasnccf]
  1> [2020-11-16T07:57:14,100][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionConditionWithFailures] waiting to start watcher, current states [Tuple [v1=node_s0, v2=STOPPED], Tuple [v1=node_s1, v2=STOPPED]]
  1> [2020-11-16T07:57:14,174][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionConditionWithFailures] waiting to start watcher, current states [Tuple [v1=node_s0, v2=STARTING], Tuple [v1=node_s1, v2=STARTING]]
  1> [2020-11-16T07:57:14,183][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionConditionWithFailures] waiting to start watcher, current states [Tuple [v1=node_s0, v2=STARTING], Tuple [v1=node_s1, v2=STARTING]]
  1> [2020-11-16T07:57:14,190][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionConditionWithFailures] waiting to start watcher, current states [Tuple [v1=node_s0, v2=STARTING], Tuple [v1=node_s1, v2=STARTING]]
  1> [2020-11-16T07:57:14,212][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionConditionWithFailures] waiting to start watcher, current states [Tuple [v1=node_s0, v2=STARTED], Tuple [v1=node_s1, v2=STARTED]]
  1> [2020-11-16T07:57:14,285][INFO ][o.e.x.w.a.l.ExecutableLoggingAction] [testActionConditionWithFailures] 0
  1> [2020-11-16T07:57:14,285][INFO ][o.e.x.w.a.l.ExecutableLoggingAction] [testActionConditionWithFailures] 1
  1> [2020-11-16T07:57:14,331][INFO ][o.e.c.m.MetadataCreateIndexService] [node_s1] [.ds-.watcher-history-14-000001] creating index, cause [initialize_data_stream], templates [.watch-history-14], shards [1]/[0]
  1> [2020-11-16T07:57:14,332][INFO ][o.e.c.r.a.AllocationService] [node_s1] updating number_of_replicas to [1] for indices [.ds-.watcher-history-14-000001]
  1> [2020-11-16T07:57:14,334][INFO ][o.e.c.m.MetadataCreateDataStreamService] [node_s1] adding data stream [.watcher-history-14] with write index [.ds-.watcher-history-14-000001] and backing indices []
  1> [2020-11-16T07:57:14,424][INFO ][o.e.x.i.IndexLifecycleTransition] [node_s1] moving index [.ds-.watcher-history-14-000001] from [null] to [{"phase":"new","action":"complete","name":"complete"}] in policy [watch-history-ilm-policy]
  1> [2020-11-16T07:57:14,594][INFO ][o.e.c.m.MetadataMappingService] [node_s1] [.ds-.watcher-history-14-000001/7HGmejYgTKKp61VUJwEU5A] update_mapping [_doc]
  1> [2020-11-16T07:57:14,834][INFO ][o.e.c.r.a.AllocationService] [node_s1] current.health="GREEN" message="Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[.ds-.watcher-history-14-000001][0]]])." previous.health="YELLOW" reason="shards started [[.ds-.watcher-history-14-000001][0]]"
  1> [2020-11-16T07:57:14,887][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionConditionWithFailures] [#testActionConditionWithFailures]: clearing watcher state
  1> [2020-11-16T07:57:14,889][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionConditionWithFailures] waiting to stop watcher, current states [Tuple [v1=node_s0 (0), v2=STARTED], Tuple [v1=node_s1 (0), v2=STARTED]]
  1> [2020-11-16T07:57:14,924][INFO ][o.e.x.w.WatcherService   ] [node_s0] stopping watch service, reason [watcher manually marked to shutdown by cluster state update]
  1> [2020-11-16T07:57:14,924][INFO ][o.e.x.w.WatcherLifeCycleService] [node_s0] watcher has stopped
  1> [2020-11-16T07:57:14,931][INFO ][o.e.x.w.WatcherService   ] [node_s1] stopping watch service, reason [watcher manually marked to shutdown by cluster state update]
  1> [2020-11-16T07:57:14,934][INFO ][o.e.x.w.WatcherLifeCycleService] [node_s1] watcher has stopped
  1> [2020-11-16T07:57:14,937][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionConditionWithFailures] waiting to stop watcher, current states [Tuple [v1=node_s0 (0), v2=STOPPED], Tuple [v1=node_s1 (0), v2=STOPPED]]
  1> [2020-11-16T07:57:14,938][INFO ][o.e.x.d.a.DeleteDataStreamTransportAction] [node_s1] removing data stream [.watcher-history-14]
  1> [2020-11-16T07:57:14,938][INFO ][o.e.c.m.MetadataDeleteIndexService] [node_s1] [.ds-.watcher-history-14-000001/7HGmejYgTKKp61VUJwEU5A] deleting index
  1> [2020-11-16T07:57:15,004][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionConditionWithFailures] [HistoryActionConditionTests#testActionConditionWithFailures]: cleaning up after test
  1> [2020-11-16T07:57:15,005][INFO ][o.e.t.InternalTestCluster] [testActionConditionWithFailures] Clearing active scheme time frozen, expected healing time 0s
  1> [2020-11-16T07:57:15,046][INFO ][o.e.c.m.MetadataDeleteIndexService] [node_s1] [.watches/G2-QCQ5eSoGSxLDkP3yP4g] deleting index
  1> [2020-11-16T07:57:15,046][INFO ][o.e.c.m.MetadataDeleteIndexService] [node_s1] [sasnccf/ftuL9UHUQT-dR8LYb2-AbA] deleting index
  1> [2020-11-16T07:57:15,150][INFO ][o.e.c.m.MetadataIndexTemplateService] [node_s1] removing template [random_index_template]
  1> [2020-11-16T07:57:15,182][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionConditionWithFailures] [HistoryActionConditionTests#testActionConditionWithFailures]: cleaned up after test
  1> [2020-11-16T07:57:15,183][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionConditionWithFailures] after test
  1> [2020-11-16T07:57:15,246][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionCondition] before test
  1> [2020-11-16T07:57:15,246][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionCondition] [HistoryActionConditionTests#testActionCondition]: setting up test
  1> [2020-11-16T07:57:15,257][WARN ][o.e.c.m.MetadataIndexTemplateService] [node_s1] legacy template [random_index_template] has index patterns [*] matching patterns from existing composable templates [.watch-history-14,.triggered_watches,.slm-history,.watches] with patterns (.watch-history-14 => [.watcher-history-14*],.triggered_watches => [.triggered_watches*],.slm-history => [.slm-history-5*],.watches => [.watches*]); this template [random_index_template] may be ignored in favor of a composable template at index creation time
  1> [2020-11-16T07:57:15,258][INFO ][o.e.c.m.MetadataIndexTemplateService] [node_s1] adding template [random_index_template] for index patterns [*]
  1> [2020-11-16T07:57:15,301][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionCondition] [HistoryActionConditionTests#testActionCondition]: all set up test
  1> [2020-11-16T07:57:15,312][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionCondition] [HistoryActionConditionTests#testActionCondition]: freezing time on nodes
  1> [2020-11-16T07:57:15,313][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionCondition] waiting to stop watcher, current states [Tuple [v1=node_s0 (0), v2=STOPPED], Tuple [v1=node_s1 (0), v2=STOPPED]]
  1> [2020-11-16T07:57:15,319][DEPRECATION][o.e.d.c.m.MetadataCreateIndexService] [node_s1] data_stream.dataset="deprecation.elasticsearch" data_stream.namespace="default" data_stream.type="logs" ecs.version="1.6" key="index_name_starts_with_dot" message="index name [.watches] starts with a dot '.', in the next major version, index names starting with a dot are reserved for hidden indices and system indices"
  1> [2020-11-16T07:57:15,326][INFO ][o.e.c.m.MetadataCreateIndexService] [node_s1] [.watches] creating index, cause [api], templates [.watches], shards [1]/[0]
  1> [2020-11-16T07:57:15,327][INFO ][o.e.c.r.a.AllocationService] [node_s1] updating number_of_replicas to [1] for indices [.watches]
  1> [2020-11-16T07:57:15,426][DEPRECATION][o.e.d.c.m.MetadataCreateIndexService] [node_s1] data_stream.dataset="deprecation.elasticsearch" data_stream.namespace="default" data_stream.type="logs" ecs.version="1.6" key="index_name_starts_with_dot" message="index name [.triggered_watches] starts with a dot '.', in the next major version, index names starting with a dot are reserved for hidden indices and system indices"
  1> [2020-11-16T07:57:15,432][INFO ][o.e.c.m.MetadataCreateIndexService] [node_s1] [.triggered_watches] creating index, cause [api], templates [.triggered_watches], shards [1]/[1]
  1> [2020-11-16T07:57:15,545][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionCondition] waiting to start watcher, current states [Tuple [v1=node_s0, v2=STOPPED], Tuple [v1=node_s1, v2=STOPPED]]
  1> [2020-11-16T07:57:15,648][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionCondition] waiting to start watcher, current states [Tuple [v1=node_s0, v2=STARTED], Tuple [v1=node_s1, v2=STARTING]]
  1> [2020-11-16T07:57:15,652][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionCondition] waiting to start watcher, current states [Tuple [v1=node_s0, v2=STARTED], Tuple [v1=node_s1, v2=STARTING]]
  1> [2020-11-16T07:57:15,658][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionCondition] waiting to start watcher, current states [Tuple [v1=node_s0, v2=STARTED], Tuple [v1=node_s1, v2=STARTED]]
  1> [2020-11-16T07:57:15,664][INFO ][o.e.x.w.WatcherService   ] [node_s0] reloading watcher, reason [new local watcher shard allocation ids], cancelled [0] queued tasks
  1> [2020-11-16T07:57:15,676][INFO ][o.e.x.w.WatcherService   ] [node_s1] reloading watcher, reason [new local watcher shard allocation ids], cancelled [0] queued tasks
  1> [2020-11-16T07:57:15,681][INFO ][o.e.c.r.a.AllocationService] [node_s1] current.health="GREEN" message="Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[.triggered_watches][0]]])." previous.health="YELLOW" reason="shards started [[.triggered_watches][0]]"
  1> [2020-11-16T07:57:15,694][INFO ][o.e.x.w.t.ScheduleTriggerEngineMock] [testActionCondition] not executing watch [testActionCondition] on this scheduler because it is paused
  1> [2020-11-16T07:57:15,704][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionCondition] [#testActionCondition]: clearing watcher state
  1> [2020-11-16T07:57:15,707][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionCondition] waiting to stop watcher, current states [Tuple [v1=node_s0 (0), v2=STARTED], Tuple [v1=node_s1 (0), v2=STARTED]]
  1> [2020-11-16T07:57:15,738][INFO ][o.e.x.w.WatcherService   ] [node_s0] stopping watch service, reason [watcher manually marked to shutdown by cluster state update]
  1> [2020-11-16T07:57:15,739][INFO ][o.e.x.w.WatcherLifeCycleService] [node_s0] watcher has stopped
  1> [2020-11-16T07:57:15,744][INFO ][o.e.x.w.WatcherService   ] [node_s1] stopping watch service, reason [watcher manually marked to shutdown by cluster state update]
  1> [2020-11-16T07:57:15,744][INFO ][o.e.x.w.WatcherLifeCycleService] [node_s1] watcher has stopped
  1> [2020-11-16T07:57:15,750][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionCondition] waiting to stop watcher, current states [Tuple [v1=node_s0 (0), v2=STOPPED], Tuple [v1=node_s1 (0), v2=STOPPED]]
  1> [2020-11-16T07:57:15,753][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionCondition] [HistoryActionConditionTests#testActionCondition]: cleaning up after test
  1> [2020-11-16T07:57:15,753][INFO ][o.e.t.InternalTestCluster] [testActionCondition] Clearing active scheme time frozen, expected healing time 0s
  1> [2020-11-16T07:57:15,783][INFO ][o.e.c.m.MetadataDeleteIndexService] [node_s1] [.watches/ySvJvDASTtyq8a3A9-oPuA] deleting index
  1> [2020-11-16T07:57:15,783][INFO ][o.e.c.m.MetadataDeleteIndexService] [node_s1] [.triggered_watches/XnnlRSQhQPCmQ4RUoB7kcg] deleting index
  1> [2020-11-16T07:57:15,843][INFO ][o.e.c.m.MetadataIndexTemplateService] [node_s1] removing template [random_index_template]
  1> [2020-11-16T07:57:15,882][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionCondition] [HistoryActionConditionTests#testActionCondition]: cleaned up after test
  1> [2020-11-16T07:57:15,884][INFO ][o.e.x.w.h.HistoryActionConditionTests] [testActionCondition] after test
  2> REPRODUCE WITH: ./gradlew ':x-pack:plugin:watcher:internalClusterTest' --tests "org.elasticsearch.xpack.watcher.history.HistoryActionConditionTests.testActionCondition" -Dtests.seed=3837630D03B63CCA -Dtests.security.manager=true -Dtests.locale=es-DO -Dtests.timezone=America/Argentina/Catamarca -Druntime.java=11
```

Another, I think:
https://gradle-enterprise.elastic.co/s/5xk7xvoozwnv2/tests/:x-pack:plugin:watcher:internalClusterTest/org.elasticsearch.xpack.watcher.history.HistoryActionConditionTests/testActionConditionWithFailures#1

```
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8
:x-pack:plugin:spatial:check
:x-pack:plugin:sql:qa:jdbc:no-sql:v7.10.1#bwcTest

REPRODUCE WITH: ./gradlew ':x-pack:plugin:watcher:internalClusterTest' --tests "org.elasticsearch.xpack.watcher.history.HistoryActionConditionTests.testActionConditionWithFailures" -Dtests.seed=966A8826BBA6FD00 -Dtests.security.manager=true -Dtests.locale=id-ID -Dtests.timezone=America/Indiana/Vincennes -Druntime.java=14

org.elasticsearch.xpack.watcher.history.HistoryActionConditionTests > testActionConditionWithFailures FAILED
    java.lang.AssertionError: 
    Expected: is <1L>
         but: was <2L>
        at __randomizedtesting.SeedInfo.seed([966A8826BBA6FD00:25DD7064282D84B8]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.watcher.history.HistoryActionConditionTests.testActionConditionWithFailures(HistoryActionConditionTests.java:154)


```

Another on master: https://gradle-enterprise.elastic.co/s/rpo5nn6hpjhce

```
java.lang.AssertionError: 
Expected: is one of {<1L>, <2L>}
     but: was <3L>
	at __randomizedtesting.SeedInfo.seed([FC7F91D4EDB37D62:CD7F3618BDE10E4]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.elasticsearch.xpack.watcher.history.HistoryActionConditionTests.testActionCondition(HistoryActionConditionTests.java:206)
```

--- a/x-pack/plugin/watcher/src/internalClusterTest/java/org/elasticsearch/xpack/watcher/history/HistoryActionConditionTests.java
+++ b/x-pack/plugin/watcher/src/internalClusterTest/java/org/elasticsearch/xpack/watcher/history/HistoryActionConditionTests.java
@@ -18,6 +18,7 @@ import org.elasticsearch.xpack.core.watcher.client.WatchSourceBuilder;
 import org.elasticsearch.xpack.core.watcher.condition.Condition;
 import org.elasticsearch.xpack.core.watcher.condition.ExecutableCondition;
 import org.elasticsearch.xpack.core.watcher.execution.ExecutionState;
+import org.elasticsearch.xpack.core.watcher.history.HistoryStoreField;
 import org.elasticsearch.xpack.core.watcher.input.Input;
 import org.elasticsearch.xpack.core.watcher.transport.actions.put.PutWatchRequestBuilder;
 import org.elasticsearch.xpack.watcher.condition.CompareCondition;
@@ -92,7 +93,6 @@ public class HistoryActionConditionTests extends AbstractWatcherIntegrationTestC
     /**
      * A hard failure is where an exception is thrown by the script condition.
      */
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/94836")
     @SuppressWarnings("unchecked")
     public void testActionConditionWithHardFailures() throws Exception {
         final String id = "testActionConditionWithHardFailures";
@@ -115,9 +115,14 @@ public class HistoryActionConditionTests extends AbstractWatcherIntegrationTestC
         assertWatchWithMinimumActionsCount(id, ExecutionState.EXECUTED, 1);
 
         // only one action should have failed via condition
-        final SearchResponse response = searchHistory(SearchSourceBuilder.searchSource().query(termQuery("watch_id", id)));
-        assertThat(response.getHits().getTotalHits().value, is(oneOf(1L, 2L)));
+        assertBusy(() -> {
+            // Watcher history is now written asynchronously, so we check this in an assertBusy
+            ensureGreen(HistoryStoreField.DATA_STREAM);
+            final SearchResponse response = searchHistory(SearchSourceBuilder.searchSource().query(termQuery("watch_id", id)));
+            assertThat(response.getHits().getTotalHits().value, is(oneOf(1L, 2L)));
+        });
 
+        final SearchResponse response = searchHistory(SearchSourceBuilder.searchSource().query(termQuery("watch_id", id)));
         final SearchHit hit = response.getHits().getAt(0);
         final List<Object> actions = getActionsFromHit(hit.getSourceAsMap());
 
@@ -144,7 +149,6 @@ public class HistoryActionConditionTests extends AbstractWatcherIntegrationTestC
         }
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/94836")
     @SuppressWarnings("unchecked")
     public void testActionConditionWithFailures() throws Exception {
         final String id = "testActionConditionWithFailures";
@@ -160,9 +164,14 @@ public class HistoryActionConditionTests extends AbstractWatcherIntegrationTestC
         assertWatchWithMinimumActionsCount(id, ExecutionState.EXECUTED, 1);
 
         // only one action should have failed via condition
-        final SearchResponse response = searchHistory(SearchSourceBuilder.searchSource().query(termQuery("watch_id", id)));
-        assertThat(response.getHits().getTotalHits().value, is(oneOf(1L, 2L)));
+        assertBusy(() -> {
+            // Watcher history is now written asynchronously, so we check this in an assertBusy
+            ensureGreen(HistoryStoreField.DATA_STREAM);
+            final SearchResponse response = searchHistory(SearchSourceBuilder.searchSource().query(termQuery("watch_id", id)));
+            assertThat(response.getHits().getTotalHits().value, is(oneOf(1L, 2L)));
+        });
 
+        final SearchResponse response = searchHistory(SearchSourceBuilder.searchSource().query(termQuery("watch_id", id)));
         final SearchHit hit = response.getHits().getAt(0);
         final List<Object> actions = getActionsFromHit(hit.getSourceAsMap());
 
@@ -189,7 +198,6 @@ public class HistoryActionConditionTests extends AbstractWatcherIntegrationTestC
     }
 
     @SuppressWarnings("unchecked")
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/65064")
     public void testActionCondition() throws Exception {
         final String id = "testActionCondition";
         final List<ExecutableCondition> actionConditions = new ArrayList<>();
@@ -211,8 +219,14 @@ public class HistoryActionConditionTests extends AbstractWatcherIntegrationTestC
         assertWatchWithMinimumActionsCount(id, ExecutionState.EXECUTED, 1);
 
         // all actions should be successful
+        assertBusy(() -> {
+            // Watcher history is now written asynchronously, so we check this in an assertBusy
+            ensureGreen(HistoryStoreField.DATA_STREAM);
+            final SearchResponse response = searchHistory(SearchSourceBuilder.searchSource().query(termQuery("watch_id", id)));
+            assertThat(response.getHits().getTotalHits().value, is(oneOf(1L, 2L)));
+        });
+
         final SearchResponse response = searchHistory(SearchSourceBuilder.searchSource().query(termQuery("watch_id", id)));
-        assertThat(response.getHits().getTotalHits().value, is(oneOf(1L, 2L)));
 
         final SearchHit hit = response.getHits().getAt(0);
         final List<Object> actions = getActionsFromHit(hit.getSourceAsMap());

